include/domain-types.h

```diff
@@ -121,6 +121,28 @@ struct Tag
 
 typedef std::vector<Tag> Tags;
 
+struct Payload
+{
+    Payload(const std::string& fileName)
+        : fileName(fileName)
+        , data(nullptr)
+        , dataSize(0)
+    {
+    }
+
+    Payload(const void* data, size_t dataSize, const std::string& mimeType)
+        : data(data)
+        , dataSize(dataSize)
+        , mimeType(mimeType)
+    {
+    }
+
+    std::string fileName;
+    const void* data;
```

Comment:
We don't transfer ownership. Caller own the data. Copying also means memory allocation and deallocation. We want to avoid those.
In case we really need to do a copy, it can be done inside method implementation, no need to made the choice in the interface.